cmd/operator/main.go

```diff
@@ -393,6 +393,35 @@ func run(fs *flag.FlagSet) int {
 		}
 	}
 
+	// If Prometheus Agent runs in DaemonSet mode, check if
+	// the operator has proper RBAC permissions on the DaemonSet resource.
+	if cfg.Gates.Enabled(operator.PrometheusAgentDaemonSetFeature) {
+		daemonSetGroupVersion := schema.GroupVersion{
+			Group:   "apps",
+			Version: "v1",
+		}
+
+		allowed, errs, err := k8sutil.IsAllowed(ctx,
+			kclient.AuthorizationV1().SelfSubjectAccessReviews(),
+			cfg.Namespaces.PrometheusAllowList.Slice(),
+			k8sutil.ResourceAttribute{
+				Group:    daemonSetGroupVersion.Group,
+				Version:  daemonSetGroupVersion.Version,
+				Resource: "daemonset",
+				Verbs:    []string{"get", "list", "watch"},
```

Comment:
I'd already account for write operations.
```suggestion
				Verbs:    []string{"get", "list", "watch", "create", "update", "delete"},
```

Reply:
Why does DaemonSet recource have these, but Prometheus Agent resource doesn't have these?

Reply:
The operator doesn't need to create/update/delete PrometheusAgent objects.

Reply:
Ah ok, those are users' roles right?

Reply:
yes only users need to modify PrometheusAgents, the operator only modifies the underlying objects such as statefulsets, secrets, configmaps and daemonsets (with your work).